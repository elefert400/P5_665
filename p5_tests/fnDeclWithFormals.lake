bool foo(int x, bool b){}
int main(){
    int myX;
    bool myB;
    myX = 1;
    myB = false;
    foo(1, true);
    foo(myX, myB);
}