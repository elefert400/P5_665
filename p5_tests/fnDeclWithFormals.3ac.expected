[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN foo LOCALS]
x (formal)
b (formal)
[END foo LOCALS]
enter foo
getin 1 x
getin 2 b
lbl_0: leave foo
[BEGIN main LOCALS]
myX (local)
myB (local)
tmp0 (tmp)
tmp1 (tmp)
[END main LOCALS]
enter main
myX := 1
myB := 0
setin 1 1
setin 2 1
call foo
getout 1 tmp0
setin 1 myX
setin 2 myB
call foo
getout 1 tmp1
lbl_1: EXIT